test/functional/test_framework/test_framework.py

```diff
@@ -275,6 +274,17 @@ def setup_nodes(self):
         self.add_nodes(self.num_nodes, extra_args)
         self.start_nodes()
         self.import_deterministic_coinbase_privkeys()
+        if not self.setup_clean_chain:
+            for n in self.nodes:
+                assert_equal(n.getblockchaininfo()["blocks"], 199)
+            self.log.debug('Generate a block with current time to finalize the cache and assert we are out of IBD')
+            block_hash = self.nodes[0].generate(1)[0]
+            block = self.nodes[0].getblock(blockhash=block_hash, verbosity=0)
+            for n in self.nodes:
+                n.submitblock(block)
+                chain_info = n.getblockchaininfo()
```

Comment:
There's no race here since `submitblock()` will only return after `ActivateBestChain()`, but if we ever make `ProcessNewBlock()` multithreaded, then there could be a race if `submitblock()` returns before the best chain has been activated.

I don't think you need to do anything here, but you could add a comment or a `wait_until()` if you wanted to make this clear to readers.